packages/webapp/components/invite/AISearchInvite.tsx

```diff
@@ -0,0 +1,132 @@
+import { ProfileImageLink } from '@dailydotdev/shared/src/components/profile/ProfileImageLink';
+import KeyIcon from '@dailydotdev/shared/src/components/icons/Key';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+import { AuthTriggers } from '@dailydotdev/shared/src/lib/auth';
+import { useMutation } from 'react-query';
+import { acceptFeatureInvitation } from '@dailydotdev/shared/src/graphql/features';
+import { useRouter } from 'next/router';
+import Logo, { LogoPosition } from '@dailydotdev/shared/src/components/Logo';
+import { useFeature } from '@dailydotdev/shared/src/components/GrowthBookProvider';
+import { feature } from '@dailydotdev/shared/src/lib/featureManagement';
+import { SearchExperiment } from '@dailydotdev/shared/src/lib/featureValues';
+import { useActions } from '@dailydotdev/shared/src/hooks/useActions';
+import { ActionType } from '@dailydotdev/shared/src/graphql/actions';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import {
+  ApiErrorResult,
+  DEFAULT_ERROR,
+} from '@dailydotdev/shared/src/graphql/common';
+import { useToastNotification } from '@dailydotdev/shared/src/hooks/useToastNotification';
+import { useAnalyticsContext } from '@dailydotdev/shared/src/contexts/AnalyticsContext';
+import { AnalyticsEvent } from '@dailydotdev/shared/src/lib/analytics';
+import { ReferralCampaignKey } from '@dailydotdev/shared/src/hooks';
+import { JoinPageProps } from './common';
+
+export function AISearchInvite({
+  referringUser,
+  redirectTo,
+  token,
+}: JoinPageProps): ReactElement {
+  const router = useRouter();
+  const search = useFeature(feature.search);
+  const { completeAction } = useActions();
+  const { trackEvent } = useAnalyticsContext();
+  const { displayToast } = useToastNotification();
+  const { user, refetchBoot, showLogin } = useAuthContext();
+  const {
+    mutateAsync: onAcceptMutation,
+    isLoading,
+    isSuccess,
+  } = useMutation(acceptFeatureInvitation, {
+    onSuccess: async () => {
+      await Promise.all([
+        completeAction(ActionType.AcceptedSearch),
+        refetchBoot(),
+      ]);
+      router.push(redirectTo);
+    },
+    onError: (err: ApiErrorResult) => {
+      const message = err?.response?.errors?.[0]?.message;
+      displayToast(message ?? DEFAULT_ERROR);
+    },
+  });
+
+  const handleAcceptClick = () => {
+    const handleAccept = () =>
+      onAcceptMutation({
+        token,
+        referrerId: referringUser.id,
+        feature: ReferralCampaignKey.Search,
+      });
+
+    if (!user) {
+      return showLogin(AuthTriggers.SearchReferral, {
+        onLoginSuccess: handleAccept,
+        onRegistrationSuccess: handleAccept,
+      });
+    }
+
+    // since in the page view, query params are tracked automatically,
+    // we don't need to send the params here explicitly
+    trackEvent({ event_name: AnalyticsEvent.AcceptInvitation });
+
+    return handleAccept();
```

Comment:
I think this should be 
```suggestion
    return handleAccept;
```

This is most likely the reason we did not see this working in our e2e tests?

Reply:
Right. Let's commit this one.